storage-proofs-core/src/util.rs

```diff
@@ -177,6 +184,46 @@ pub fn default_rows_to_discard(leafs: usize, arity: usize) -> usize {
     }
 }
 
+// Based on
+// https://stackoverflow.com/questions/73145503/iterator-for-reading-file-chunks/73145594#73145594
+/// Chunks the given reader to the given size.
+///
+/// If the end is reached and there are a few bytes left, that don't fill a full chunk, those bytes
+/// are returned.
+pub struct ChunkIterator<R> {
```

Comment:
Will this be used exclusively in `filecoin-proofs`? If so, it may make sense for it to be defined there, rather than in `storage-proofs-core`.

Reply:
Very good point. I've moved it to `filecoin-proofs`.